driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -653,29 +626,22 @@ public void onSet(Connection connection, Message.Response response, long latency
                                 retryCount, queryState, queryStateRef.get());
                         return;
                     }
-
                     connection.release();
-
-                    switch (response.type) {
-                        case RESULT:
-                            if (((Responses.Result) response).kind == Responses.Result.Kind.PREPARED) {
-                                logger.debug("Scheduling retry now that query is prepared");
-                                retry(true, null);
-                            } else {
-                                logError(connection.address, new DriverException("Got unexpected response to prepare message: " + response));
-                                retry(false, null);
-                            }
-                            break;
-                        case ERROR:
-                            logError(connection.address, new DriverException("Error preparing query, got " + response));
-                            if (metricsEnabled())
-                                metrics().getErrorMetrics().getOthers().inc();
-                            retry(false, null);
-                            break;
-                        default:
-                            // Something's wrong, so we return but we let setFinalResult propagate the exception
-                            SpeculativeExecution.this.setFinalResult(connection, response);
-                            break;
+                    if (response.type == RESULT && ((Responses.Result) response).kind == PREPARED) {
```

Comment:
Here only the success case is bypassing the retry policy, all other (error) cases go through it.